omod/src/main/webapp/resources/htmlforms/PatientReferral.xml

```diff
@@ -80,21 +80,61 @@
+                var transferOutName = getPatientTransferEncounterTypes("e305d98a-d6a2-45ba-ba2a-682b497ce27c");
                 var data1 = jq.ajax({
                     dataType: "json",
-                    url: "http://localhost:8080/openmrs/ws/rest/v1/encounter?patient=990d60c5-1966-45b0-9222-3322f7a59317&amp;v=custom%3A(display%2Cobs%3A(concept%3A(conceptId)%2CvalueText%2CvalueNumeric%2CvalueDatetime%2CvalueCoded%3A(conceptId)))&amp;encounterType=8d5b27bc-c2cc-11de-8d13-0010c6dffd0f&amp;limit=1",
+                    url: '/'+OPENMRS_CONTEXT_PATH +"/ws/rest/v1/encounter?patient="+ patientuuid +"&amp;v=custom%3A(display%2Cobs%3A(concept%3A(conceptId)%2CvalueText%2CvalueNumeric%2CvalueDatetime%2CvalueCoded%3A(conceptId)))&amp;encounterType=8d5b27bc-c2cc-11de-8d13-0010c6dffd0f&amp;limit=1",
                     data: "",
                     success: "success"
                 });
 
                 var data2 = jq.ajax({
                     dataType: "json",
-                    url: "http://localhost:8080/openmrs/ws/rest/v1/encounter?patient=990d60c5-1966-45b0-9222-3322f7a59317&amp;v=custom%3A(display%2Cobs%3A(concept%3A(conceptId)%2CvalueText%2CvalueNumeric%2CvalueDatetime%2CvalueCoded%3A(conceptId)))&amp;encounterType=8d5b2be0-c2cc-11de-8d13-0010c6dffd0ff&amp;limit=1",
+                    url: '/'+ OPENMRS_CONTEXT_PATH +"/ws/rest/v1/encounter?patient="+patientuuid +"&amp;v=custom%3A(display%2Cobs%3A(concept%3A(conceptId)%2CvalueText%2CvalueNumeric%2CvalueDatetime%2CvalueCoded%3A(conceptId)))&amp;encounterType=8d5b2be0-c2cc-11de-8d13-0010c6dffd0ff&amp;limit=1",
                     data: "",
                     success: "success"
                 });
 
+                var transferDate = getValue('encounterDate.value');
+
+                jq('#encounter_type').change(function(){
+                    transferDate= getValue('encounterDate.value');
+                    var type = jq('#encounter_type option:selected').text();
+                    if(type==transferInName){
+                        setValue('99160.value',transferDate);
+                        setValue('99165.value','');
+                    }else if(type==transferOutName){
+                       setValue('99165.value',transferDate);
+                        setValue('99160.value','');
+                    }
+                });
+
+                jq('#encounterDate').change(function(){
+                    transferDate= getValue('encounterDate.value');
+                    var type = jq('#encounter_type option:selected').text();
+                    if(type==transferInName){
+                        setValue('99160.value',transferDate);
+                        setValue('99165.value','');
+                    }else if(type==transferOutName){
+                        setValue('99165.value',transferDate);
+                        setValue('99160.value','');
+                    }
+                });
+
+                jq('#encounterDate').each(function(){
```

Comment:
@dbaluku you do not need an each for the encounter date as it is only one as it is referenced by ID - or am I missing something, we should not have multiple IDs referring to the same HTML element

Reply:
Done